plugins/editqml/qml/PaletteControls.qml

```diff
@@ -699,60 +788,33 @@ QtObject{
         palettes.data = filterOutPalettes(
             palettes.data,
             paletteGroup.palettesOpened,
-            mode === PaletteControls.PaletteListMode.ObjectContainer || mode === PaletteControls.PaletteListMode.NodeEditor
+            mode === PaletteControls.PaletteListMode.ObjectContainer
         )
 
         if (!palettes.data || palettes.data.length === 0) return null
 
-        var paletteList = createPaletteListView(listParent ? listParent : null, theme.selectableListView)
-        paletteList.model = palettes.data
-
-        var palListBox = null
-        if (mode !== PaletteControls.PaletteListMode.NodeEditor){
-
-            var pane = container.parent
-            if ( container.pane )
-                pane = container.pane
-            while (pane && pane.objectName !== "editor" && pane.objectName !== "objectPalette"){
-                pane = pane.parent
-                if ( pane && pane.pane && pane.pane.objectName === 'objectPalette' ){
-                    pane = pane.pane
-                }
+        var pane = container.parent
+        if ( container.pane )
+            pane = container.pane
+        while (pane && pane.objectName !== "editor" && pane.objectName !== "objectPalette"){
+            pane = pane.parent
+            if ( pane && pane.pane && pane.pane.objectName === 'objectPalette' ){
+                pane = pane.pane
             }
-
-            var paneCoords = pane.mapGlobalPosition()
-
-            palListBox  = lk.layers.editor.environment.createEditorBox(
-                paletteList,
-                aroundBox,
-                Qt.point(paneCoords.x, paneCoords.y - 35),
-                lk.layers.editor.environment.placement.bottom
-            )
-
-            palListBox.color = 'transparent'
-
-        } else {
-            container.objectGraph.paletteListOpened = true
         }
+        var paneCoords = pane.mapGlobalPosition()
 
-        paletteList.forceActiveFocus()
-
-
-        paletteList.cancelledHandler = function(){
-            paletteList.focus = false
-            paletteList.model = null
-            paletteList.destroy()
-            if (mode !== PaletteControls.PaletteListMode.NodeEditor) palListBox.destroy()
-            else {
-                container.objectGraph.paletteListOpened = false
-                container.objectGraph.activateFocus()
-            }
+        var editorBoxParams = {
+            aroundRect: aroundBox,
+            panePosition: Qt.point(paneCoords.x, paneCoords.y - 35),
+            relativePlacement: lk.layers.editor.environment.placement.bottom
         }
 
-        paletteList.selectedHandler = function(index){
-            paletteList.focus = false
-            paletteList.model = null
+        var cancelledHandler = function(paletteListBox){
+            paletteListBox.destroy()
```

Comment:
I think we need to destroy the `paletteList` as well:

```js
paletteListBox.item.destroy()
```

Reply:
Please check the very first lines in PaletteControls.qml where we're doing these boilerplate things.